### PR TITLE
[storage] Support get mvcc by ts

### PR DESCRIPTION
**Pull request by cfzjywxk**
Support mvcc get ts read for debugging usage

```
curl http://127.0.0.1:10080/mvcc/txn/405179368526053377/test/t1
```
Returns the transaction commits/rollbacks up to the input startTs

### CODE REVIEW

**Comment by cfzjywxk** [comment]
@coocood @bobotu PTAL

**Comment by cfzjywxk** [comment]
@coocood @bobotu  PTAL

File: tikv/mvcc.go [hunk 1]
```diff
@@ -953,42 +953,72 @@ func (store *MVCCStore) getRollbackMvcc(rawkey []byte, ts uint64,
...
-	// try to find correspond txn with input startTs using oltIter
 	var item *badger.Item
+	// try to find using iter
+	startKey := reqCtx.regCtx.startKey
+	endKey := reqCtx.regCtx.endKey
+	iter := reader.GetIter()
+	for iter.Seek(startKey); iter.Valid(); iter.Next() {
+		curItem := iter.Item()
+		curKey := curItem.Key()
+		if bytes.Compare(curKey, endKey) > 0 {
```
Comment (coocood) [inline tikv/mvcc.go]: >=

File: tikv/mvcc.go [hunk 2]
```diff
@@ -953,42 +953,72 @@ func (store *MVCCStore) getRollbackMvcc(rawkey []byte, ts uint64,
...
+	for oldIter.Seek(oldStartKey); oldIter.Valid(); oldIter.Next() {
 		curItem := oldIter.Item()
-		if len(curItem.UserMeta()) > 0 {
-			meta := mvcc.DBUserMeta(curItem.UserMeta())
-			if meta.StartTS() == startTs {
-				item = curItem
-				break
-			}
+		curKey := curItem.Key()
+		if bytes.Compare(curKey, oldEndKey) > 0 {
```
Comment (coocood) [inline tikv/mvcc.go]: `>=`

File: tikv/mvcc.go [hunk 3]
```diff
@@ -953,42 +953,72 @@ func (store *MVCCStore) getRollbackMvcc(rawkey []byte, ts uint64,
...
+		if err != nil {
+			return nil, nil, err
+		}
+		return res, resKey, nil
+	}
+	// try to find correspond txn with input startTs using oltIter
+	// startKey = "t" and oldKey = "u" cannot be mapped using EncodeOldKey directly
+	var oldStartKey []byte
+	var oldEndKey []byte
+	if IsRecordKey(startKey) {
```
Comment (coocood) [inline tikv/mvcc.go]: We can `EncodeOldKey` no matter if it is record key.

File: tikv/dbreader/db_reader.go [hunk 4]
```diff
@@ -51,14 +51,24 @@ func (r *DBReader) GetMvccInfoByKey(key []byte, isRowKey bool, mvccInfo *kvrpcpb
...
+		err = r.getOldKeysWithMeta(oldKey, isRowKey, mvccInfo)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// GetMvccInfoByOldKey fills MvccInfo reading old keys from db
+func (r *DBReader) GetMvccInfoByOldKey(oldKey []byte, isRowKey bool, mvccInfo *kvrpcpb.MvccInfo) error {
```
Comment (coocood) [inline tikv/dbreader/db_reader.go]: This is not used.

Comment (cfzjywxk) [reply tikv/dbreader/db_reader.go]: Removed

**Comment by coocood** [comment]
LGTM

**Pull request merged**